src/cards/vacuum-card/vacuum-card.ts

```diff
@@ -93,36 +97,53 @@ export class VacuumCard extends MushroomBaseCard implements LovelaceCard {
-                        ?rtl=${rtl}
-                        .appearance=${appearance}
-                        @action=${this._handleAction}
-                        .actionHandler=${actionHandler({
-                            hasHold: hasAction(this._config.hold_action),
-                            hasDoubleClick: hasAction(this._config.double_tap_action),
-                        })}
+                      ?rtl=${rtl}
+                      .appearance=${appearance}
+                      @action=${this._handleAction}
+                      .actionHandler=${actionHandler({
+                          hasHold: hasAction(this._config.hold_action),
+                          hasDoubleClick: hasAction(this._config.double_tap_action),
+                      })}
                     >
                         ${picture ? this.renderPicture(picture) : this.renderIcon(entity, icon)}
                         ${this.renderBadge(entity)}
                         ${this.renderStateInfo(entity, appearance, name)};
                     </mushroom-state-item>
                     ${isCommandsControlVisible(entity, commands)
-                        ? html`
-                              <div class="actions" ?rtl=${rtl}>
-                                  <mushroom-vacuum-commands-control
-                                      .hass=${this.hass}
-                                      .entity=${entity}
-                                      .commands=${commands}
-                                      .fill=${appearance.layout !== "horizontal"}
-                                  >
-                                  </mushroom-vacuum-commands-control>
-                              </div>
-                          `
-                        : null}
+                      ? html`
+                          <div class="actions" ?rtl=${rtl}>
+                              <mushroom-vacuum-commands-control
+                                .hass=${this.hass}
+                                .entity=${entity}
+                                .commands=${commands}
+                                .fill=${appearance.layout !== "horizontal"}
+                              >
+                              </mushroom-vacuum-commands-control>
+                          </div>
+                      `
+                      : null}
                 </mushroom-card>
             </ha-card>
         `;
     }
 
+    protected renderIcon(entity: HassEntity, icon: string): TemplateResult {
+        let iconStyle = {};
+
+        return html`
+            <mushroom-shape-icon
+                slot="icon"
+                class=${classMap({
+                    returning: isReturningHome(entity) && Boolean(this._config?.icon_animation),
+                    cleaning: isCleaning(entity) && Boolean(this._config?.icon_animation),
+                })}
+                style=${styleMap(iconStyle)}
+                .disabled=${!isActive(entity) && !isReturningHome(entity)}
```

Comment:
```suggestion
                .disabled=${!isActive(entity)}
```
Should be enough as returning home is now an active state with https://github.com/piitaya/lovelace-mushroom/pull/855

Reply:
Nice!